e2e/node/e2e.test.ts

```diff
@@ -7,313 +23,269 @@ import {
   revokeAccessGrant,
   getFile,
 } from "../../src/index";
-import { Session } from "@inrupt/solid-client-authn-node";
-import {
-  isVerifiableCredential,
-  VerifiableCredential,
-} from "@inrupt/solid-client-vc";
-import { config } from "dotenv-flow";
+import { getTestingEnvironment } from "../e2e-setup";
 
-// Load environment variables from .env.test.local if available:
-config({
-  path: __dirname,
-  // In CI, actual environment variables will overwrite values from .env files.
-  // We don't need warning messages in the logs for that:
-  silent: process.env.CI === "true",
-});
-
-type OidcIssuer = string;
-type VcService = string;
-type Requestor = string;
-type ResourceOwner = string;
-type ClientId = string;
-type ClientSecret = string;
-type AuthDetails = [
-  OidcIssuer,
-  VcService,
-  Requestor,
-  ResourceOwner,
-  ClientId,
-  ClientSecret,
-  ClientId,
-  ClientSecret
-];
-// Instructions for obtaining these credentials can be found here:
-// https://github.com/inrupt/solid-client-authn-js/blob/1a97ef79057941d8ac4dc328fff18333eaaeb5d1/packages/node/example/bootstrappedApp/README.md
-const serversUnderTest: AuthDetails[] = [
-  // dev-next.inrupt.com:
-  [
-    // Cumbersome workaround, but:
-    // Trim `https://` from the start of these URLs,
-    // so that GitHub Actions doesn't replace them with *** in the logs.
-    process.env.E2E_TEST_DEV_NEXT_IDP_URL!.replace(/^https:\/\//, ""),
-    process.env.E2E_TEST_DEV_NEXT_VC_SERVICE!.replace(/^https:\/\//, ""),
-    process.env.E2E_TEST_DEV_NEXT_REQUESTOR!.replace(/^https:\/\//, ""),
-    process.env.E2E_TEST_DEV_NEXT_RESOURCE_OWNER!.replace(/^https:\/\//, ""),
-    process.env.E2E_TEST_ESS_REQUESTOR_CLIENT_ID!,
-    process.env.E2E_TEST_ESS_REQUESTOR_CLIENT_SECRET!,
-    process.env.E2E_TEST_ESS_RESOURCE_OWNER_CLIENT_ID!,
-    process.env.E2E_TEST_ESS_RESOURCE_OWNER_CLIENT_SECRET!,
-  ],
-];
-
-const SHARED_FILE_IRI =
-  "https://storage.dev-next.inrupt.com/eb2f327b-7bb4-4ba2-9b4b-678a4d7e3551/somefile.txt";
+const {
+  idp: oidcIssuer,
+  environment,
+  requestor,
+  resourceOwner,
+  vcProvider,
+} = getTestingEnvironment();
 
 // This is the content of the file uploaded manually at SHARED_FILE_IRI.
 const SHARED_FILE_CONTENT = "Some content.\n";
 
-describe.each(serversUnderTest)(
-  "Access grant client end-to-end tests authenticated to [%s], issuing from [%s] for [%s]",
-  (
-    oidcIssuerDisplay,
-    vcServiceDisplay,
-    requestorDisplay,
-    resourceOwnerDisplay,
-    requestorClientId,
-    requestorClientSecret,
-    resourceOwnerClientId,
-    resourceOwnerClientSecret
-  ) => {
-    const oidcIssuer = new URL(`https://${oidcIssuerDisplay}`).href;
-    const vcService = new URL(`https://${vcServiceDisplay}`).href;
-    const requestor = new URL(`https://${requestorDisplay}`).href;
-    const resourceOwner = new URL(`https://${resourceOwnerDisplay}`).href;
+describe(`End-to-end access grant tests for environment [${environment}}]`, () => {
```

Comment:
Isn't this self-documenting via the matrix runner?

<img width="904" alt="Screenshot 2022-01-31 at 17 03 41" src="https://user-images.githubusercontent.com/30827/151828731-ffc0754c-af0a-48e3-930f-92a426cd4206.png">

We could get rid of all the environment validation type logic and instead rely on the runner to be showing the environment? We could also do a check in getEnvironment that if not `process.env.CI` then print out the environment you're executing on as developer diagnostics.